src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -143,37 +144,34 @@ public static Set<Course> parseCourses(Collection<String> courses) throws ParseE
      *
      * @throws ParseException if the given {@code tutorial} is invalid.
      */
-    public static Tutorial parseTutorial(Set<Course> courseList, String tutorial) throws ParseException {
-        requireNonNull(tutorial);
-        String trimmedTutorial = tutorial.trim();
-        if (!Tutorial.isValidTutorialString(trimmedTutorial)) {
+    public static Tutorial parseTutorial(Set<Course> courseSet, String tutorialString) throws ParseException {
+        requireNonNull(tutorialString);
+        String trimmedTutorialString = tutorialString.trim();
+        if (!Tutorial.isValidTutorialString(trimmedTutorialString)) {
             throw new ParseException(Tutorial.MESSAGE_CONSTRAINTS);
         }
 
-        Course relevantCourse = null;
-        String[] courseTutorialName = Tutorial.splitCourseTutorialName(trimmedTutorial);
+
+        // Check if in the format of COURSECODE/TUTORIALCODE
+        String[] courseTutorialName = Tutorial.splitCourseTutorialName(trimmedTutorialString);
         if (courseTutorialName == null || courseTutorialName.length <= 0) {
             throw new ParseException(Tutorial.MESSAGE_CONSTRAINTS);
         }
 
-        for (Course course : courseList) {
-            if (course.courseName.equals(courseTutorialName[0])) {
-                relevantCourse = course;
-            }
-        }
-        if (relevantCourse == null) {
-            String givenCoursesString = courseList.stream()
-                .map((course) -> course.courseName)
-                .reduce(
-                        "", (current, next) -> current + next.toString() + " "
-                       );
-            givenCoursesString = String.format("[%s]", givenCoursesString);
-            throw new ParseException(String.format(
-                        Tutorial.INVALID_COURSE_MESSAGE, courseTutorialName[0], givenCoursesString
-                        ));
-        }
-
-        return new Tutorial(relevantCourse, trimmedTutorial);
+        // Get the relevant course if it exists.
+        Optional<Course> relevantCourse = Tutorial.findMatchingCourse(courseSet, trimmedTutorialString);
+        Tutorial parsedTutorial = relevantCourse.map((course) -> {
+            return new Tutorial(course, trimmedTutorialString);
+        }).orElseThrow(() -> {
+            // Prepare the error message; we should format the invalid course message with
+            // the concatenation of all course strings.
+            String allCoursesString = courseSet.stream()
+                .map((course) -> course.getCourseName())
+                .reduce("", (current, next) -> current + next.toString() + "  ");
+            return new ParseException(String.format(Tutorial.INVALID_COURSE_MESSAGE, courseTutorialName[0], allCoursesString));
```

Comment:
Maybe can devide this into 2 lines so that every line is not more than 120 chars.